installers-conda/build_installers.py

```diff
@@ -0,0 +1,402 @@
+        return
+
+    from PIL import Image
+
+    logo_path = SPYREPO / "img_src" / "spyder.png"
+    logo = Image.open(logo_path, "r")
+
+    if installer_type in ("exe", "all"):
+        sidebar = Image.new("RGBA", (164, 314), (0, 0, 0, 0))
+        sidebar.paste(logo.resize((101, 101)), (32, 180))
+        output = DIST / "spyder_164x314.png"
+        sidebar.save(output, format="png")
+
+        banner = Image.new("RGBA", (150, 57), (0, 0, 0, 0))
+        banner.paste(logo.resize((44, 44)), (8, 6))
+        output = DIST / "spyder_150x57.png"
+        banner.save(output, format="png")
+
+    if installer_type in ("pkg", "all"):
+        _logo = Image.new("RGBA", logo.size, "WHITE")
+        _logo.paste(logo, mask=logo)
+        background = Image.new("RGBA", (1227, 600), (0, 0, 0, 0))
+        background.paste(_logo.resize((148, 148)), (95, 418))
+        output = DIST / "spyder_1227x600.png"
+        background.save(output, format="png")
+
+
+def _get_condarc():
+    # we need defaults for tensorflow and others on windows only
+    defaults = "- defaults" if WINDOWS else ""
+    prompt = "[spyder]({default_env}) "
+    contents = dedent(
+        f"""
+        channels:  #!final
+          - spyder-ide
+          - conda-forge
+          {defaults}
+        repodata_fns:  #!final
+          - repodata.json
+        auto_update_conda: false  #!final
+        notify_outdated_conda: false  #!final
+        channel_priority: strict  #!final
+        env_prompt: '{prompt}'  #! final
+        """
+    )
+    # the undocumented #!final comment is explained here
+    # https://www.anaconda.com/blog/conda-configuration-engine-power-users
+    file = DIST / "condarc"
+    file.write_text(contents)
+
+    return str(file)
+
+
+def _definitions():
+    condarc = _get_condarc()
+    definitions = {
+        "name": APP,
+        "company": "Spyder-IDE",
+        "reverse_domain_identifier": "org.spyder-ide.Spyder",
+        "version": SPYVER,
+        "channels": [
+            "napari/label/bundle_tools",
+            "spyder-ide",
+            "conda-forge",
+        ],
+        "conda_default_channels": ["conda-forge"],
+        "specs": [
+            "python",
+            "conda",
+            "mamba",
+            "pip",
+        ],
+        "installer_filename": OUTPUT_FILE.name,
+        "initialize_by_default": False,
+        "license_file": str(RESOURCES / "bundle_license.rtf"),
+        "extra_envs": {
+            f"spyder-{SPYVER}": {
+                "specs": [k + v for k, v in specs.items()],
+            },
+        },
+        "menu_packages": [
+            "spyder",
+        ],
+        "extra_files": {
+            str(RESOURCES / "bundle_readme.md"): "README.txt",
+            condarc: ".condarc",
+        },
+    }
+    if not args.no_local:
+        definitions["channels"].insert(0, "local")
+
+    if LINUX:
+        definitions["default_prefix"] = os.path.join(
+            "$HOME", ".local", INSTALLER_DEFAULT_PATH_STEM
```

Comment:
I think we should validate that `~/.local` exists at this point and if it doesn't, then we should create it. We can't take that for granted given the immense variety of Linux distros.

Reply:
I believe this is done by the installer at install time on the local machine. This `default_prefix` specifies where, on the local machine, the environment should be installed, but this script `build_installers.py` is not run on the local machine. So checking for `~.local` here will only check if it is present on the build machine.

Reply:
Ok, I see. No problem with that then.